### ISSUE TITLE
Flag for custom search paths

### ISSUE THREAD

**Issue by Globegitter** [nathan/pax#68]
I have been trying to integrate pax with [bazel](https://bazel.build/) and first experiments look really promising - only issue I ran into, if you have files generated by bazel (e.g. a vue/ts/jsx transformed file), bazel puts this into different directory root, so a relative import statement of `import a from './lib/a';` would not be found at that path, but it might be at `./bazel-out/k8s-fastbuild/lib/a.js` or similar. I wounder if it would be possible to add a flag without too much difficult, e.g. `--search-paths /path/to/dir` and then in addition to where pax would look originally for a file, it will also look for it in the specified paths, if not found. So for the above import statement, it might also look for it also at `/path/to/dir/lib/a.js`. Does that make sense?

**Comment by nathan** [comment]
If bazel is processing all of your source code, wouldn't the full module tree be available at `./bazel-out/k8s-fastbuild/`? In that case, you could simply run Pax on the path to the main module in the output directory and be done. If you need `node_modules` there and it isn't, you should be able to just symlink it.

I think the option you suggested doesn't make much sense, since relative paths are universally resolved with respect to the directory of the source file they appear in, and I don't know of any bundlers that let you change that behavior. It might make sense to let the user specify a map of *absolute* path substitutions, e.g. `lib/a.js` => `/path/to/dir/lib/a.js`, so that any reference that resolves to `<cwd>/lib/a.js` resolves to `/path/to/dir/lib/a.js`, but I'm not sure how useful that would be.

**Comment by Globegitter** [comment]
Yeah I agree, it is one of the behaviours of bazel that has confused quite a few users but bazel has a concept of "source" and "generated" files and it source files it keeps in the original place and generated files are being put into `bazel-out`. I do have a workaround in place right now where bazel copies all source files next to the generated files, but that adds quite a performance hit once the file add up. The other and probably cheapest option would be to just symlink the bazel-out directory back to the root and that would indeed work, but I wanted to see if it is possible to have native support.

For example typescript does provide such an option, they call it `rootDirs`: https://www.typescriptlang.org/docs/handbook/module-resolution.html

The map solution sounds quite interesting as well, but either way can also understand if you feel this is too narrow of a use-case.

**Comment by rope-hmg** [comment]
It would be nice if there was a root directory type flag. The application that I work on is many thousands of files large and we currently use webpack's `resolve modules` feature. It allows us to import files from either `node_modules` or from our `src` directories without using `'../../../../../../file_name'` type paths. 

We have the following in our webpack config:

    resolve: {
        modules: [SOURCE_PATH, NODE_MODULES_PATH],
    },

Thanks

**Comment by rista404** [comment]
I think this would be helpful when dealing with monorepos (in my case yarn workspaces).

**Comment by Akiyamka** [comment]
@rope-hmg it can be done more native way without  webpack. 
https://www.npmjs.com/package/link-module-alias
